1-python-question.py

```diff
@@ -23,29 +23,64 @@
     "applicants": ["Amy", "Ally", "David", "Brendan", "Zoho"]
 }
 
+allapplicants = request["applicants"]
 
 def processRequest(request):
-    # Your code here
-    return
 
+    result =  {
+   "successfulApplicants": [],
+   "bannedApplicants": [],
+   "totalCost": 0,
+   "tickets": []
 
-print(processRequest(request))
+ }
+
+   # 1. filter call banned visitors
+  
+    if (not(len(allapplicants) == 0)):
+       def checkBan(i):
+          return ((i in bannedVisitors) == True)
+       
+
+       result["bannedApplicants"] = list(filter(checkBan, allapplicants))
+   
+   # 2. check applicants' membership status
+       def checkAllow(i):
+           return ((i in bannedVisitors) == False)
+
+       allowedVisitors = list(filter(checkAllow, allapplicants))
 
-# {
-#   successfulApplicants:
-#   bannedApplicatns:
-#   totalCost:
-#   tickets: [
-#       {
-#            "name": ,
-#            "membershipStatus": ,
-#            "price":
-#       }, ....
-#   ]
-#
-# }
+       def checkMember(i):
+              return memberStatus.get(str(i)) == True
 
+       def checkVisitor(i):
+              return memberStatus.get(str(i)) == False or (not(i in memberStatus.keys()))
```

Comment:
you can take advantage of .get() behavior and do
```
              return memberStatus.get(str(i))
```
or the inverse of this depending on your needs. consider reading up what values python considers falsy.

also the logic for checkVisitor is more complicated than it has to be, but it works nonetheless